Makefile

```diff
@@ -12,6 +12,14 @@
 ## See the License for the specific language governing permissions and
 ## limitations under the License.
 
+## Convention regarding this Makefile
+# NOTE: This makefile has very coarse-grained targets. The same targets get
+# executed by circleci as well. Please do not add any fine-grained targets,
```

Comment:
I'm not sure what circleci has to do with anything.  There's also builds in other contexts like prow and cloud builder.  If the build is sufficiently robust and hermetic then the differences in environments become less of an issue.  There's some benefit to be able to look at the cited dependencies for a target--if they get lumped together as a group then when somebody removes a dependency for one they're less apt to review all the other components in the group to see if the dependency can be removed for the entire group.  In my experience the main pitfall with higher precision is that occasionally someone discovers that one of them doesn't build from a clean state (because some required dependencies weren't cited).

Reply:
I think the comment should be "same targets get executed in CI/CD systems" - they should not be specific to circleci.

We can have fine-grained targets for development - but we should have some discipline, maybe have some naming conventions ( like cicd/foo ) ?

Reply:
BTW - it would be good for a 'help' target that lists the targets that CICD execute (presubmit, etc), and maybe match the name of the target with the job in the CICD, to be extra clear. And make it clear the goal is to have consistency/reproductibility between developer-local targets and what CICD presubmits/etc are doing ( regardless of which CICD is used).